app/models/adyen_notification.rb

```diff
@@ -84,5 +101,43 @@ def actions
       map(&:downcase)
   end
 
+  # https://docs.adyen.com/display/TD/Notification+fields
+  def modification_event?
+    [ CANCELLATION,
+      REFUND,
+      CANCEL_OR_REFUND,
+      CAPTURE,
+      CAPTURE_FAILED,
+      REFUND_FAILED,
+      REFUNDED_REVERSED
+    ].member? self.event_code
+  end
+
+  def normal_event?
+    AUTHORISATION == self.event_code
+  end
+
+  def bank_transfer?
+    self.payment_method.match(/^bankTransfer/)
```

Comment:
You could eliminate the need for a match object and do `self.payment_method =~ /^bankTransfer/`; no match will return nil.